File: leetcode/lc-us/cpp/1833.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    int maxIceCream(vector<int> &costs, int coins) {
        sort(costs.begin(), costs.end());
        int cnt = 0;
        for (int cost: costs) {
            if (coins >= cost) {
                cnt++;
                coins -= cost;
            } else {
                break;
            }
        }
        return cnt;
    }
};

File: leetcode/lc-us/py/1833.py
from typing import List


class Solution:
    def maxIceCream(self, costs: List[int], coins: int) -> int:
        costs.sort()
        tot = 0
        for i, cost in enumerate(costs):
            tot += cost
            if tot > coins:
                return i
        return len(costs) if tot <= coins else 0
